python/lsst/ap/association/dia_object.py

```diff
@@ -190,9 +198,27 @@ def append_dia_source(self, input_dia_source_record):
         # them until we are finished adding sources.
         self._updated = False
 
-        self._dia_source_catalog.append(
-            self._dia_source_catalog.getTable().copyRecord(
-                input_dia_source_record))
+        # We need to test that schema of the source to appended lines up with
+        # the schema specified in the catalog of dia_sources associated with
+        # this dia_object. If not we attempt to access the those that overlap
+        # assuming that the schema defined in this dia_object is a subset of
+        # of the input source's schema.
+        input_schema = input_dia_source_record.getSchema()
+        if input_schema == self._dia_source_schema:
```

Comment:
This `if:else:` block seems overly complicated. Is there not a simpler way to add the record?

Reply:
Its complicated because I want to get a copy of the record as well as make sure the schemas line up. The way to do the second could possibly be taken care of by a SchemaMapper but creating the mapping may in fact result in more complex code than what you see here.